pkg/apis/options/login_url_parameters.go

```diff
@@ -0,0 +1,54 @@
+package options
```

Comment:
On first observation, this should not be in this package. The API is deliberately importing a relatively small amount of external packages to avoid dependencies on other packages.

The construction of the regexes would normally be moved closer to wherever they are being used, probably somewhere in the providers package.

Is the intention here that the configuration loading would unmarshal the regex into a re when it loads? Seems convenient, perhaps we can keep this, but is there any reason not to pass strings around until we construct the regexes in the provider construction?

How do errors manifest when a user provides an invalid regex?